### PR TITLE
Suomi.fi authentication

### PR DESCRIPTION
**Pull request by OP-Lamminen**

### CODE REVIEW

File: tunnistamo/tests/test_suomifi_authentication.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    expected_metadata = load_file('suomifi_metadata.xml')
+    assert metadata_response.status_code == 200
+    assert metadata_response.get('content-type') == 'text/xml'
+    assert metadata_response.content == expected_metadata
+
+
+@pytest.mark.django_db
+@freeze_time('2019-01-01 12:00:00', tz_offset=2)
+def test_suomifi_login_request(django_client, fixed_saml_id):
+    '''Suomi.fi use case #1'''
```
Comment (tituomin) [inline tunnistamo/tests/test_suomifi_authentication.py]: Where are these use cases documented and numbered? I'll store the reference in an appropriate place.

File: tunnistamo/settings.py [hunk 2]
```diff
@@ -383,9 +387,123 @@
 SOCIAL_AUTH_ESPOO_ADFS_KEY = ''
 SOCIAL_AUTH_ESPOO_ADFS_SECRET = None
 
-IPWARE_META_PRECEDENCE_ORDER = ('REMOTE_ADDR',)
+# Social Auth Suomi.fi values. These should be overwritten in local settings.
```
Comment (tituomin) [inline tunnistamo/settings.py]: Should *all* of these be overwritten? For example SOCIAL_AUTH_SUOMIFI_SP_EXTRA ?
Do the test settings contain enough of a documentation (in the form of an example) of what should be configured and how, or would we need a separate documentation?

File: tunnistamo/tests/test_suomifi_authentication.py (hunk 1 above)
Comment (OP-Lamminen) [reply tunnistamo/tests/test_suomifi_authentication.py]: https://palveluhallinta.suomi.fi/fi/sivut/tunnistus/kayttoonotto/asiakastestiymparisto

File: tunnistamo/settings.py (hunk 2 above)
Comment (OP-Lamminen) [reply tunnistamo/settings.py]: Settings have been reordered and comments are now a bit more verbose.

**Pull request merged**